core/src/main/java/org/eclipse/krazo/core/ViewResponseFilter.java

```diff
@@ -211,7 +211,10 @@ static String appendExtensionIfRequired(String viewName, String defaultExtension
         }
 
         String resultView = viewName;
-        if (!viewName.contains(".")) {
+        if (defaultExtension.contains(".")) {
```

Comment:
Maybe I'm missing something obvious here, but isn't this incorrect? Wouldn't this always add the default extension to the view name, even if the view name already contains a `.`?

I would have expected something like this instead:

```java
if (!viewName.contains(".")) {
  resultView += "." + defaultExtension.replaceAll("^\.", "");
}
```

Or if we want to avoid the regex:

```java
if (!viewName.contains(".")) {
  if(!defaultExtension.startsWith(".")) {
    resultView += ".";
  }
  resultView += defaultExtension);
}
```

Reply:
I'm not sure but I think I wrote test for this and it worked correct🤔 I'll double check it

Reply:
You're right, this would introduce a bug. I added tests and fixed the case when a view already has a view assigned (at least we assume it in case it contains a dot).